### PR TITLE
add new reward system and PPO training script

### PR DESCRIPTION
**Pull request by tigerwlin**
major changes:
1. added a new type of reward system
2. refactored the reward handler class such that switch between different reward system can be done by a one line change of the class name
3. added training script for PPO

### CODE REVIEW

**Comment by tigerwlin** [comment]
@olegsinyavskiy @ghostFaceKillah please review

File: bc_gym_planning_env/envs/base/action.py [hunk 1]
```diff
@@ -2,11 +2,29 @@
...
-import attr
-import numpy as np
 
-
-@attr.s
 class Action(object):
     """ Object representing an 'action' - a motion primitive to execute in the environment """
-    command = attr.ib(type=np.ndarray)
+
+    def __init__(self, command=None):
```
Comment (ghostFaceKillah) [inline bc_gym_planning_env/envs/base/action.py]: Please do not remove compatibility of others while introducing new functionality.

File: bc_gym_planning_env/envs/base/env.py [hunk 2]
```diff
@@ -129,13 +130,10 @@ def __init__(self, costmap, path, params, random_param=None):
         """
         # Stateful things
         self._robot = TricycleRobot(dimensions=get_dimensions_example(params.robot_name))
-        self._reward_provider = ContinuousRewardProvider(params=params.reward_provider_params)
+        self._reward_provider = ContinuousRewardPurePursuitProvider(params=params.reward_provider_params)
```
Comment (ghostFaceKillah) [inline bc_gym_planning_env/envs/base/env.py]: Please do not change behaviour for everyone while introducing behaviour for your case. Please dependency inject RewardProvider, based on parametrization.

**Review by ghostFaceKillah — CHANGES_REQUESTED** [review]

File: bc_gym_planning_env/envs/base/env.py (hunk 2 above)
Comment (tigerwlin) [reply bc_gym_planning_env/envs/base/env.py]: fixed as suggested

File: bc_gym_planning_env/envs/base/action.py (hunk 1 above)
Comment (tigerwlin) [reply bc_gym_planning_env/envs/base/action.py]: refactored as a new class function

**Comment by tigerwlin** [comment]
conflict from after serialization resolved. please review again @ghostFaceKillah @olegsinyavskiy

File: bc_gym_planning_env/envs/base/action.py [hunk 3]
```diff
@@ -25,3 +25,29 @@ def __eq__(self, other):
 
     def __ne__(self, other):
         return not self.__eq__(other)
+
+    def processed_command(self):
```
Comment (olegsinavski) [inline bc_gym_planning_env/envs/base/action.py]: I don't know what it is, but this function looks shady.. It returns different hardcoded numbers depending on the shape and is in 'base' folder (assuming it supposed to be quite generic)

Comment (tigerwlin) [reply bc_gym_planning_env/envs/base/action.py]: agreed. Making this class can handle both discrete and continuous action seems non-trivial at this point. Maybe I should revert this change and create a separate PR to focus on solving this issue

**Comment by tigerwlin** [comment]
letting the system handle both discrete and continuous action space seems need more work than I thought. Maybe let's separate that into another PR? @olegsinyavskiy 
@ghostFaceKillah

**Pull request merged**